src/Sylius/Component/Core/Model/Shipment.php

```diff
@@ -45,4 +53,36 @@ public function setOrder(BaseOrderInterface $order = null)
 
         return $this;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStockLocation()
+    {
+        return $this->stockLocation;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setStockLocation(BaseStockLocationInterface $location = null)
+    {
+        $this->stockLocation = $location;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getShippingWeight()
+    {
+        $weight = 0;
+
+        foreach ($this->items as $item) {
+            $weight = +$item->getShippable()->getShippingWeight();
```

Comment:
This is wrong. Should be `$weight += $item->getShippable()->getShippingWeight();`